### PR TITLE
Implement working with messages guide draft

### PR DESCRIPTION
**Pull request by dmigach**
Please check out a draft for working with messages guide, I believe for complete picture it's best to run it locally

### CODE REVIEW

File: stream-chat-swift-docs/docs/guides/working-with-messages.md [hunk 1]
```diff
@@ -1,3 +1,310 @@
 ---
 title: Working with Messages
----
\ No newline at end of file
+---
+
+## Message object
+A message is represented by `ChatMessage` model.
```
Comment (VojtaStavik) [inline stream-chat-swift-docs/docs/guides/working-with-messages.md:6]: Should this also describe the message UI object? We will probably have a separate guide for customizing the message list but some basic overview would make sense here? @b-onc ?

Comment (VojtaStavik) [reply stream-chat-swift-docs/docs/guides/working-with-messages.md:6]: There's nothing like `ChatMessageGroup` 🙈

Comment (VojtaStavik) [reply stream-chat-swift-docs/docs/guides/working-with-messages.md:6]: I mean the object you see in the message list that represents a message

File: stream-chat-swift-docs/docs/guides/working-with-messages.md [hunk 2]
```diff
@@ -202,6 +199,12 @@ class MyChannelViewController: UIViewController {
 
 ## Delete a message
 
+When a user deletes a message it will be hidden for all the rest users in conversation, but it will appear for the user who deleted it like this:
+
+  <img src="../img/message-delete.png" width="50%" />
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/working-with-messages.md:207]: Are you sure this is correct? don't we show "_Message deleted_"?

File: stream-chat-swift-docs/docs/guides/working-with-messages.md [hunk 3]
```diff
@@ -4,17 +4,19 @@ title: Working with Messages
 
 ## Message object
 A message is represented by `ChatMessage` model.
+Depending on combination of its properties messages appear differently, this screenshot showcases the most common types of messages:
+
+<img src="../img/messages-showcase.png" width="40%" />
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/working-with-messages.md:9]: Can we also include an image where we show which element in the message UI corresponds to which property of message? not all, just some

File: stream-chat-swift-docs/docs/guides/working-with-messages.md [hunk 4]
```diff
@@ -274,10 +277,11 @@ There are two ways of replying a message:
   
     Initiating a thread reply takes a user into thread details screen and the resulting message will look like a normal message that is placed inside the thread. It is also possible to duplicate it to the parent channel.
   
- <img src="../img/thread-reply.png" width="20%" />
+    <img src="../img/thread-details.png" width="20%" />
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/working-with-messages.md:283]: It'd be better if the original message belonged to another user

Comment (dmigach) [reply stream-chat-swift-docs/docs/guides/working-with-messages.md:283]: Ok let's change that

**Review by b-onc — APPROVED** [review]
You can put the 2 images in the first section side-by-side, if it's too much effort just ignore :)

**Comment by dmigach** [comment]
> You can put the 2 images in the first section side-by-side, if it's too much effort just ignore :)

Good idea

**Pull request merged**